test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQServiceTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Microsoft.Azure.WebJobs.Extensions.RabbitMQ;
+using Moq;
+using RabbitMQ.Client;
+using Xunit;
+
+namespace WebJobs.Extensions.RabbitMQ.Tests
+{
+    public class RabbitMQServiceTests
+    {
+        [Theory]
+        [InlineData(null, "localhost", "queue", "guest", "guest", 5672)]
+        [InlineData("amqp://user:PASSWORD@52.175.195.81:5672", null, "queue", null, null, null)]
+        [InlineData(null, "localhost", "queue", null, null, 0)] // Should fill in "guest", "guest", 5672
+        public void Handles_Connection_Attributes_And_Options(string connectionString, string hostName, string queueName, string userName, string password, int port)
+        {
+            RabbitMQService service = new RabbitMQService(connectionString, hostName, queueName, userName, password, port);
+
+            ConnectionFactory factory = service.CreateConnectionFactory();
+            if (connectionString == null && userName == "guest")
+            {
+                Assert.Null(factory.Uri);
+                Assert.Equal("localhost", factory.HostName);
+                Assert.Equal("guest", factory.UserName);
+                Assert.Equal("guest", factory.Password);
+                Assert.Equal(5672, factory.Port);
+            }
+            else if (connectionString != null)
+            {
+                Assert.Equal(new Uri("amqp://user:PASSWORD@52.175.195.81:5672"), factory.Uri);

```

Comment:
use dummy ip

Reply:
If I do this the connection will fail - what would be the best solution for this

Reply:
Just test CreateConnectionFactory. do not create RabbitMQ Service

Reply:
Do I mock RabbitMQService? How do I test this without actually creating the connection since that's what the method does?